tests/unit_tests/service/family/test_update_family_service.py

```diff
@@ -0,0 +1,299 @@
+    family_repo_mock,
+    collection_repo_mock,
+    geography_repo_mock,
+    organisation_repo_mock,
+    metadata_repo_mock,
+    app_user_repo_mock,
+):
+    family = family_service.get("a.b.c.d")
+    assert family is not None  # needed to placate pyright
+
+    updated_family = create_family_write_dto(collections=["x.y.z.2", "col3", "col4"])
+
+    with pytest.raises(ValidationError) as e:
+        family_service.update("a.b.c.d", USER_EMAIL, updated_family)
+    expected_msg = "The import ids are invalid: ['col3', 'col4']"
+    assert e.value.message == expected_msg
+
+    assert family_repo_mock.get.call_count == 2
+    assert geography_repo_mock.get_id_from_value.call_count == 1
+    assert family_repo_mock.update.call_count == 0
+    assert app_user_repo_mock.get_org_id.call_count == 1
+    assert organisation_repo_mock.get_id_from_name.call_count == 1
+    assert metadata_repo_mock.get_schema_for_org.call_count == 1
+    assert collection_repo_mock.validate.call_count == 0
+    assert collection_repo_mock.get_org_from_collection_id.call_count == 0
+
+
+def test_update_family_raises_when_collection_missing(
+    family_repo_mock,
+    collection_repo_mock,
+    geography_repo_mock,
+    organisation_repo_mock,
+    metadata_repo_mock,
+    app_user_repo_mock,
+):
+    family = family_service.get("a.b.c.d")
+    assert family is not None  # needed to placate pyright
+
+    updated_family = create_family_write_dto()
+
+    collection_repo_mock.missing = True
+    with pytest.raises(ValidationError) as e:
+        family_service.update("a.b.c.d", USER_EMAIL, updated_family)
+    expected_msg = "One or more of the collections to update does not exist"
+    assert e.value.message == expected_msg
+
+    assert family_repo_mock.get.call_count == 2
+    assert geography_repo_mock.get_id_from_value.call_count == 1
+    assert family_repo_mock.update.call_count == 0
+    assert app_user_repo_mock.get_org_id.call_count == 1
+    assert organisation_repo_mock.get_id_from_name.call_count == 1
+    assert metadata_repo_mock.get_schema_for_org.call_count == 1
+    assert collection_repo_mock.validate.call_count == 1
+    assert collection_repo_mock.get_org_from_collection_id.call_count == 0
+
+
+def test_update_family_raises_when_collection_org_different_to_usr_org(
+    family_repo_mock,
+    collection_repo_mock,
+    geography_repo_mock,
+    organisation_repo_mock,
+    metadata_repo_mock,
+    app_user_repo_mock,
+):
+    family = family_service.get("a.b.c.d")
+    assert family is not None  # needed to placate pyright
+
+    updated_family = create_family_write_dto(collections=["x.y.z.2", "x.y.z.3"])
+
+    collection_repo_mock.invalid_org = True
+    with pytest.raises(ValidationError) as e:
+        family_service.update("a.b.c.d", USER_EMAIL, updated_family)
+    expected_msg = "Organisation mismatch between some collections and the current user"
+    assert e.value.message == expected_msg
+
+    assert family_repo_mock.get.call_count == 2
```

Comment:
Begs the question if the assertion below actually matter - as long as the ValidationError was raised :thinking:

Reply:
Ah this was helpful for debugging the tests initially - there is some funky stuff going on with the mocks in some places. Would like to keep these in for now until we've re-written our mocks if poss